### PR TITLE
[Sprint 2] List all shop

### PR DESCRIPTION
**Pull request by NguyenTienVinh**
Redmine link : https://dev.framgia.com/redmine/issues/113701
Task: List all shop
Effect: N/A

### CODE REVIEW

File: app/src/main/java/com/framgia/forder/screen/listshop/ListShopAdapter.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        mShops = new ArrayList<>();
+        if (mShops == null) {
+            return;
+        }
+        mShops.addAll(shops);
+    }
+
+    @Override
+    public ItemViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        ItemListShopBinding binding =
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/forder/screen/listshop/ListShopAdapter.java]: can use 
ItemListShopBinding binding = ItemListShopBinding.inflate(LayoutInflater.from(parent.getContext());

File: app/src/main/java/com/framgia/forder/screen/listshop/ListShopAdapter.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+public class ListShopAdapter extends BaseRecyclerViewAdapter<ListShopAdapter.ItemViewHolder> {
+    private List<Shop> mShops;
+    private OnRecyclerViewItemClickListener<Object> mItemClickListener;
+
+    protected ListShopAdapter(@NonNull Context context, List<Shop> shops) {
+        super(context);
+        mShops = new ArrayList<>();
+        if (mShops == null) {
+            return;
+        }
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/listshop/ListShopAdapter.java]: remove.
mShops initialized above, no need check not null

File: app/src/main/java/com/framgia/forder/screen/listshop/ListShopAdapter.java (hunk 1 above)
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/forder/screen/listshop/ListShopAdapter.java]: Pls create a instance variable for context.

File: app/src/main/java/com/framgia/forder/screen/listshop/ListShopAdapter.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+import com.framgia.forder.screen.mainpage.shop.ItemShopViewModel;
+import com.framgia.forder.utils.navigator.Navigator;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by ASUS on 20-04-2017.
+ */
+
+public class ListShopAdapter extends BaseRecyclerViewAdapter<ListShopAdapter.ItemViewHolder> {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/listshop/ListShopAdapter.java]: this Adapter contains 2 type item Header & content
Pls update this Adapter

File: app/src/main/java/com/framgia/forder/screen/listshop/ListShopPresenter.java [hunk 4]
```diff
@@ -8,21 +21,48 @@
...
 
     @Override
     public void onStop() {
+        mCompositeSubscription.clear();
     }
 
     @Override
     public void getListAllShop() {
-
+        Domain domain = mDomainRepository.getCurrentDomain();
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/listshop/ListShopPresenter.java]: if you call this method on ``onStart()`` please remove declare this method on ``Contract.Presenter`` class

File: app/src/main/java/com/framgia/forder/screen/listshop/ListShopAdapter.java [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+    void updateData(List<Shop> shops) {
+        if (shops == null) {
+            return;
+        }
+        mShops.clear();
+        mShops.addAll(shops);
+        notifyDataSetChanged();
+    }
+
+    private class HeaderViewHolder extends RecyclerView.ViewHolder {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/listshop/ListShopAdapter.java]: change to
```java
private static class HeaderViewHolder extends RecyclerView.ViewHolder
```

File: app/src/main/java/com/framgia/forder/screen/listshop/ListShopAdapter.java [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+            mItemClickListener = listener;
+        }
+
+        void bind(Shop shop) {
+            mBinding.setViewModel(new ItemShopViewModel(shop, mItemClickListener));
+            mBinding.executePendingBindings();
+        }
+    }
+
+    private class ContentViewHolder extends RecyclerView.ViewHolder {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/listshop/ListShopAdapter.java]: change to 
```java
private static class ContentViewHolder extends RecyclerView.ViewHolder
```

File: app/src/main/java/com/framgia/forder/screen/listshop/ListShopAdapter.java [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+
+/**
+ * Created by ASUS on 20-04-2017.
+ */
+
+public class ListShopAdapter extends BaseRecyclerViewAdapter<RecyclerView.ViewHolder> {
+    private List<Shop> mShops;
+    private static OnRecyclerViewItemClickListener<Object> mItemClickListener;
+    private static final int HEAD = 0;
+    private static final int CONTENT = 1;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/listshop/ListShopAdapter.java]: the constant variable should be put on top of the file

File: app/src/main/java/com/framgia/forder/screen/listshop/ListShopAdapter.java [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+import java.util.List;
+
+/**
+ * Created by ASUS on 20-04-2017.
+ */
+
+public class ListShopAdapter extends BaseRecyclerViewAdapter<RecyclerView.ViewHolder> {
+    private List<Shop> mShops;
+    private static OnRecyclerViewItemClickListener<Object> mItemClickListener;
+    private static final int HEAD = 0;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/listshop/ListShopAdapter.java]: change name to ``HEADER``

File: app/src/main/java/com/framgia/forder/screen/listshop/ListShopAdapter.java [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by ASUS on 20-04-2017.
+ */
+
+public class ListShopAdapter extends BaseRecyclerViewAdapter<RecyclerView.ViewHolder> {
+    private List<Shop> mShops;
+    private static OnRecyclerViewItemClickListener<Object> mItemClickListener;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/listshop/ListShopAdapter.java]: remove static

File: app/src/main/res/layout/item_header_shop.xml [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+                            android:layout_height="wrap_content"
+                            android:layout_gravity="bottom"
+                            android:maxLines="1"
+                            android:text="@{viewModel.description}"
+                            android:textColor="@color/colorAccent"
+                            android:textSize="@dimen/sp_20"
+                            />
+
+                    </LinearLayout>
+
```
Comment (daolq3012) [inline app/src/main/res/layout/item_header_shop.xml]: remove empty line

File: app/src/main/res/layout/item_header_shop.xml [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+                </LinearLayout>
+
```
Comment (daolq3012) [inline app/src/main/res/layout/item_header_shop.xml]: remove empty line

File: app/src/main/res/layout/item_header_shop.xml [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+            </FrameLayout>
+
```
Comment (daolq3012) [inline app/src/main/res/layout/item_header_shop.xml]: remove empty line

File: app/src/main/res/layout/item_header_shop.xml [hunk 13]
```diff
@@ -0,0 +1,126 @@
...
+        </LinearLayout>
+
```
Comment (daolq3012) [inline app/src/main/res/layout/item_header_shop.xml]: remove empty line

File: app/src/main/res/layout/item_header_shop.xml [hunk 14]
```diff
@@ -0,0 +1,126 @@
...
+    </android.support.v7.widget.CardView>
+
```
Comment (daolq3012) [inline app/src/main/res/layout/item_header_shop.xml]: remove empty line

File: app/src/main/res/layout/item_list_shop.xml [hunk 15]
```diff
@@ -0,0 +1,159 @@
...
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:layout_gravity="left"
+                        android:progressTint="@color/colorAccent"
+                        android:rating="@{viewModel.rating}"
+                        style="?android:attr/ratingBarStyleSmall"
+                        />
+
+                </LinearLayout>
+
```
Comment (daolq3012) [inline app/src/main/res/layout/item_list_shop.xml]: remove empty line

File: app/src/main/res/layout/item_list_shop.xml [hunk 16]
```diff
@@ -0,0 +1,159 @@
...
+            </LinearLayout>
+
```
Comment (daolq3012) [inline app/src/main/res/layout/item_list_shop.xml]: remove empty line

File: app/src/main/res/layout/item_list_shop.xml [hunk 17]
```diff
@@ -0,0 +1,159 @@
...
+        </LinearLayout>
+
```
Comment (daolq3012) [inline app/src/main/res/layout/item_list_shop.xml]: remove empty line

File: app/src/main/res/layout/item_list_shop.xml [hunk 18]
```diff
@@ -0,0 +1,159 @@
...
+    </android.support.v7.widget.CardView>
+
```
Comment (daolq3012) [inline app/src/main/res/layout/item_list_shop.xml]: remove empty line

**Pull request merged**